tests/ui/lazy-type-alias/def-site-predicates-wf.rs

```diff
@@ -0,0 +1,21 @@
+//! Ensure that we check the predicates at the definition site for well-formedness.
+#![feature(lazy_type_alias)]
+#![allow(incomplete_features)]
+
+//@ revisions: current next
+//@[next] compile-flags: -Znext-solver
+//@ ignore-compare-mode-next-solver (explicit revisions)
+
+type Alias0 = ()
+where
+    <Vec<str> as Discard>::Output:; //~ ERROR the size for values of type `str` cannot be known at compilation time
```

Comment:
Ah, this can actually be simplified to just `Vec<str>:` etc. I copy pasted that from the corresp. GCI PR where that was necessary. Lemme fix that